src/NServiceBus.Core/Recoverability/RecoverabilitySettings.cs

```diff
@@ -0,0 +1,230 @@
+        {
+            Guard.AgainstNull(nameof(custom), custom);
+
+            Settings.Set(Recoverability.PolicyOverride, custom);
+
+            return this;
+        }
+    }
+
+    /// <summary>
+    /// Provides information about the recoverability configuration.
+    /// </summary>
+    public struct RecoverabilityConfig
+    {
+        internal RecoverabilityConfig(ImmediateConfig immediateConfig, DelayedConfig delayedConfig)
+        {
+            Immediate = immediateConfig;
+            Delayed = delayedConfig;
+        }
+
+        /// <summary>
+        /// Exposes the immediate retries configuration.
+        /// </summary>
+        public ImmediateConfig Immediate { get; }
+
+        /// <summary>
+        /// Exposes the delayed retries configuration.
+        /// </summary>
+        public DelayedConfig Delayed { get; }
+    }
+
+    /// <summary>
+    /// Provides information about the immediate retries configuration.
+    /// </summary>
+    public struct ImmediateConfig
+    {
+        internal ImmediateConfig(int maxNumberOfRetries)
+        {
+            MaxNumberOfRetries = maxNumberOfRetries;
+        }
+
+        /// <summary>
+        /// Gets the configured maximum number of immediate retries.
+        /// </summary>
+        /// <remarks>Zero means no retries possible.</remarks>
+        public int MaxNumberOfRetries { get; }
+    }
+
+    /// <summary>
+    /// Provides information about the delayed retries configuration.
+    /// </summary>
+    public struct DelayedConfig
+    {
+        internal DelayedConfig(int maxNumberOfRetries, TimeSpan timeIncrease)
+        {
+            MaxNumberOfRetries = maxNumberOfRetries;
+            TimeIncrease = timeIncrease;
+        }
+
+        /// <summary>
+        /// Gets the configured maximum number of immediate retries.
+        /// </summary>
+        /// <remarks>Zero means no retries possible.</remarks>
+        public int MaxNumberOfRetries { get; }
+
+        /// <summary>
+        /// Gets the configured time of increase for individual delayed retries.
+        /// </summary>
+        public TimeSpan TimeIncrease { get; }
+    }
+
+    /// <summary>
+    /// Extension methods for recoverability which extend <see cref="EndpointConfiguration" />.
+    /// </summary>
+    public static class RecoverabilityEndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Configuration settings for recoverability.
+        /// </summary>
+        /// <param name="configuration">The endpoint configuration.</param>
+        public static RecoverabilitySettings Recoverability(this EndpointConfiguration configuration)
+        {
+            return new RecoverabilitySettings(configuration.GetSettings());
+        }
+    }
+
+    /// <summary>
+    /// Configuration settings for immediate retries.
+    /// </summary>
+    public class ImmediateRetriesSettings : ExposeSettings
```

Comment:
I guess we could have `Config` folder with all types split out to separate files. I'm not sure it's worth it though.